src/Authentication/Authentication.Core/Authenticator.cs

```diff
@@ -77,36 +106,84 @@ public static async Task<IAuthContext> AuthenticateAsync(IAuthContext authContex
                 }
 
                 JwtHelpers.DecodeJWT(httpRequestMessage.Headers.Authorization?.Parameter, account, ref authContext);
-                return authContext;
+                return (authContext, new AuthError(AuthErrorType.None, null));
             }
             catch (AuthenticationException authEx)
             {
-                if ((authEx.InnerException is TaskCanceledException) && cancellationToken.IsCancellationRequested)
+                //Interactive Authentication Failure: Could Not Open Browser, fallback to DeviceAuth
+                if (IsUnableToOpenWebPageError(authEx))
+                {
```

Comment:
> There are two scenarios.
> 
> * A user may not be in an interactive session. - We detect this by the call to `IsUserInteractive()`
> * A User may be in an interactive session but for some reason (Browser not installed for example) the browser fails to open.  - This can only be known through the exception thrown. Thus we need to fallback to DeviceCode auth in this scenario. Thats where the recursion in the catch block comes in.

@finsharp, both scenarios can be handled in this method once the [proposed method signature](https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/618/files/8e7f5bb2879a9403c6d3ac115e7c59c422e95103#r616269503) is implemented. You can let `fallBackWarning` only log the warning message [without checking the type](https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/618/files/8e7f5bb2879a9403c6d3ac115e7c59c422e95103#diff-6765c82282b378bf2f4edc2b785424abe653b006678a02f6b9f4cc0317036bc0R217), and let the caller check the type before invoking the callback action. e.g.
```suggestion
                {
                    if (fallBackWarning != null)
                        fallBackWarning();
```